src/titlepane/TitlePane.ts

```diff
@@ -0,0 +1,127 @@
+ * @property onRequestOpen      Called when the title of a closed pane is clicked
+ * @property open               If true the pane is opened and content is visible
+ * @property title              Title to display above the content
+ */
+export interface TitlePaneProperties extends ThemeableProperties {
+	closeable?: boolean;
+	headingLevel?: number;
+	onRequestClose?(titlePane: TitlePane): void;
+	onRequestOpen?(titlePane: TitlePane): void;
+	open?: boolean;
+	title: string;
+};
+
+export const TitlePaneBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class TitlePane extends TitlePaneBase<TitlePaneProperties> {
+	private _contentId = uuid();
+	private _titleId = uuid();
+
+	private _afterRender(element: HTMLElement) {
+		// Conditionally adjust top margin. Done manually instead of through Maquette
+		// so the underlying DOM is accessible, as we need to know the content height.
+		// Put in a rAF to push this operation to the next tick, otherwise
+		// element.offsetHeight can be incorrect (e.g. before styling is applied)
+		// Note that this will go away when meta support is added to widget-core
+		requestAnimationFrame(() => {
+			const { open = true } = this.properties;
+			const height = element.offsetHeight;
+			element.style.marginTop = open ? '0px' : `-${ height }px`;
+		});
+	}
+
+	private _onTitleClick() {
+		this._toggle();
+	}
+
+	private _onTitleKeyUp(event: KeyboardEvent) {
+		const {keyCode } = event;
+
+		if (keyCode === Keys.Enter || keyCode === Keys.Space) {
+			this._toggle();
+		}
+	}
+
+	private _toggle() {
+		const {
+			closeable = true,
+			onRequestClose,
+			onRequestOpen,
+			open = true
+		} = this.properties;
+
+		if (!closeable) {
+			return;
+		}
+
+		if (open) {
+			onRequestClose && onRequestClose(this);
+		}
+		else {
+			onRequestOpen && onRequestOpen(this);
+		}
+	}
+
+	protected onElementCreated(element: HTMLElement, key: string) {
+		key === 'content' && this._afterRender(element);
+	}
+
+	protected onElementUpdated(element: HTMLElement, key: string) {
+		key === 'content' && this._afterRender(element);
+	}
+
+	render(): DNode {
+		const {
+			closeable = true,
+			headingLevel,
+			open = true,
+			title
+		} = this.properties;
+
+		return v('div', {
+			classes: this.classes(css.root)
+		}, [
+			v('div', {
+				'aria-level': headingLevel ? String(headingLevel) : null,
+				classes: this.classes(css.title, closeable ? css.closeable : null),
+				onclick: this._onTitleClick,
+				onkeyup: this._onTitleKeyUp,
+				role: 'heading'
+			}, [
+				v('div', {
+					'aria-controls': this._contentId,
+					'aria-disabled': closeable ? null : 'true',
+					'aria-expanded': String(open),
+					id: this._titleId,
+					role: 'button',
+					tabIndex: closeable ? 0 : -1
+				}, [ title ])
+			]),
+			v('div', {
+				'aria-labelledby': this._titleId,
+				classes: this.classes(css.content),
+				id: this._contentId,
+				key: 'content'
```

Comment:
Sorry, I missed something last time :P. Could we add `aria-hidden: open ? null : 'true'` to this?